src/server.cpp

```diff
@@ -369,12 +381,10 @@ Server::~Server()
 
 		infostream << "Server: Saving environment metadata" << std::endl;
 		m_env->saveMeta();
-	}
 
-	// Stop threads
-	if (m_thread) {
-		stop();
-		delete m_thread;
+		// Note that this also deletes and saves the map.
+		delete m_env;
+		m_env = nullptr;
```

Comment:
setting m_env to nullptr in destructor means we have a scope issue somewhere in the destruction. Which code part uses m_env after that and will break ? (it means we should find a wait to split some objects at a moment)

Reply:
this is just good practice to not leave dangling pointers around